qa/os/src/test/java/org/elasticsearch/packaging/util/Docker.java

```diff
@@ -291,6 +291,22 @@ public static void copyFromContainer(Path from, Path to) {
 
             return super.getScriptCommand("docker exec --user elasticsearch:root --tty " + containerId + " " + script);
         }
+
+        @Override
+        public Result run(String script) {
+            try {
+                return super.run(script);
+            } catch (ShellException e) {
+                try {
+                    final Shell.Result dockerLogs = getContainerLogs();
+                    throw new ShellException(
+                        e.getMessage() + "\n\nContainer stdout: " + dockerLogs.stdout + "\n\nContainer stderr: " + dockerLogs.stderr
+                    );
+                } catch (ShellException logsException) {
```

Comment:
Why are we catching an exception we just threw on the line above? It seems like we are adding an additional stack frame, why?